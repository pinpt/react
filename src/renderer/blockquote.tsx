import { registerNode, recurseIntoChildren, NodeProps } from './register';

const BlockQuote = ({ node }: NodeProps) => {
	return <blockquote>{recurseIntoChildren(node)}</blockquote>;
};

registerNode('blockquote', (node) => <BlockQuote node={node} />);
