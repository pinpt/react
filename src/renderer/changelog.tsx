import { ReactYouTubeLite as Youtube } from 'react-youtube-lite';
import { getFileUrl } from '../lib/file_metadata';
import { slugifyContent } from '../lib/string';
import { ChangelogCoverMedia, ChangelogMediaType } from '../lib/types';
import { Content as Doc } from './';

const ImageMedia = ({ src, title = '' }: { src: string; title?: string }) => {
	return (
		<div className="image">
			<img src={src} alt={title} />
		</div>
	);
};

const VideoMedia = ({ src }: { src: string }) => {
	return (
		<div className="video">
			<video controls src={src} />
		</div>
	);
};

const YoutubeMediaSSR = ({ id, poster = 'hqdefault' }: { id: string; poster?: string }) => {
	// the component isn't SSR friendly so this is basically used to sort of mimic it for generated code
	const posterUrl = `https://i.ytimg.com/vi/${id}/${poster}.jpg`;
	return (
		<>
			{poster && <link rel="preload" href={posterUrl} as="image" />}
			<div
				className="youtube ryt-lite embed-responsive aspect-ratio-16/9"
				style={{ backgroundImage: `url(${posterUrl})` }}
			>
				<div className="lty-playbtn" data-url={`https://www.youtube-nocookie.com/embed/${id}?autoplay=1`} />
			</div>
		</>
	);
};

const YoutubeMedia = ({
	id,
	metadata,
	staticMode,
}: {
	id: string;
	metadata?: Record<string, any>;
	staticMode?: true;
}) => {
	return (
		<div className="youtube">
			{!staticMode ? (
				<Youtube url={`https://www.youtube.com/watch?v=${id}`} poster={metadata?.poster} />
			) : (
				<YoutubeMediaSSR id={id} poster={metadata?.poster} />
			)}
		</div>
	);
};

const Media = ({ media, title, staticMode }: { media?: ChangelogCoverMedia; title?: string; staticMode?: true }) => {
	if (!media) {
		return <></>;
	}
	let content: React.ReactElement;
	switch (media.type) {
		case ChangelogMediaType.None: {
			return <></>;
		}
		case ChangelogMediaType.Image: {
			content = <ImageMedia src={getFileUrl(media.value)} title={title} />;
			break;
		}
		case ChangelogMediaType.Video: {
			content = <VideoMedia src={getFileUrl(media.value)} />;
			break;
		}
		case ChangelogMediaType.Youtube: {
			content = <YoutubeMedia id={media.value} metadata={media.metadata} staticMode={staticMode} />;
			break;
		}
	}
	return <div className="media-container">{content}</div>;
};

interface ChangelogProps {
	id: string;
	title?: string;
	content?: any;
	coverMedia?: ChangelogCoverMedia;
	limit?: number;
	divider?: boolean;
	staticMode?: true;
}

const ChangelogLog = (props: ChangelogProps) => {
	return (
		<article className="changelog notebook-editor read-only">
			{props.coverMedia && (
				<section className="covermedia">
					<Media media={props.coverMedia} title={props.title} staticMode={props.staticMode} />
				</section>
			)}
			<section className="ProseMirror">
				<Doc node={props.content} limit={props.limit} />
				{props.limit && props.content.content?.length > props.limit && (
					<div className="continue">
						<a href={slugifyContent(props.id, props.title)}>Continue Reading â†’</a>
					</div>
				)}
			</section>
			{props.divider && <hr className="divider" />}
		</article>
	);
};

export default ChangelogLog;
