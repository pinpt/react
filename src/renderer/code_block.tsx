import { Prism as SyntaxHighlighter } from 'react-syntax-highlighter';
import { twilight } from 'react-syntax-highlighter/dist/cjs/styles/prism';
import { NodeProps, registerNode } from './register';

const CodeBlock = ({ node }: NodeProps) => {
	return (
		<SyntaxHighlighter
			language={node.attrs.language}
			style={twilight}
			customStyle={{
				background: 'black',
				fontFamily: 'ui-monospace',
				borderWidth: '0px',
				borderRadius: '.25em',
			}}
		>
			{node.content?.[0].text ?? ''}
		</SyntaxHighlighter>
	);
};

registerNode('code_block', (node) => <CodeBlock node={node} />);
