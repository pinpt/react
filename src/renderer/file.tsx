import {
	faDownload, faFile, faFileAlt, faFileArchive, faFileAudio, faFileCode, faFileExcel, faFilePdf,
	faFilePowerpoint, faFileVideo, faFileWord
} from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { NodeProps, registerNode } from './register';

export const iconForType = (type = '') => {
	if (/pdf/.test(type)) {
		return faFilePdf;
	}
	if (/zip|tar/.test(type)) {
		return faFileArchive;
	}
	if (/video/.test(type)) {
		return faFileVideo;
	}
	if (/audio/.test(type)) {
		return faFileAudio;
	}
	if (/excel|spreadsheet/.test(type)) {
		return faFileExcel;
	}
	if (/powerpoint/.test(type)) {
		return faFilePowerpoint;
	}
	if (/word/.test(type)) {
		return faFileWord;
	}
	if (/css|csv|graphql|html|java|json|php|python|typescript|xml/.test(type)) {
		return faFileCode;
	}
	if (/text/.test(type)) {
		return faFileAlt;
	}
	return faFile;
};

export const formattedSize = (size = 0) => {
	const kB = size / 1000;
	if (kB < 10) {
		return `${Math.round(kB * 10) / 10} KB`;
	}
	if (kB < 1000) {
		return `${Math.round(kB)} KB`;
	}
	const MB = kB / 1000;
	return `${Math.round(MB * 10) / 10} MB`;
};

const File = ({ node }: NodeProps) => {
	const { description, filename, publicUrl, size, type } = node.attrs;

	const onDownload = () => {
		if (publicUrl) {
			const u = new URL(publicUrl);
			u.searchParams.set('filename', filename);
			u.searchParams.set('disposition', 'attachment');
			const a = document.createElement('a');
			a.style.display = 'none';
			a.href = u.toString();
			// the filename you want
			a.download = filename;
			document.body.appendChild(a);
			a.click();
			a.remove();
		}
	};

	return (
		<div className="file">
			<div className="container">
				<FontAwesomeIcon icon={iconForType(type)} />
				<div className="details">
					{publicUrl ? (
						<a href={publicUrl} target="_blank" rel="noopener noreferrer" className="filename">
							{filename}
						</a>
					) : (
						<div className="filename">{filename}</div>
					)}
					{description && <div className="description">{description}</div>}
					<div className="size">{formattedSize(size)}</div>
				</div>
				<div className="options">
					{publicUrl && (
						<div className="option group" onClick={onDownload}>
							<FontAwesomeIcon icon={faDownload} />
							<span>Download</span>
						</div>
					)}
				</div>
			</div>
		</div>
	);
};

registerNode('file', (node) => <File node={node} />);
