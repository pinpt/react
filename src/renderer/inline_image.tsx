import { extractImageMetadataFromFileID } from '../lib/file_metadata';
import { NodeProps, registerNode } from './register';

const InlineImage = ({ node }: NodeProps) => {
	const { size } = extractImageMetadataFromFileID(node.attrs?.src);
	const width = size?.width;
	const height = size?.height;

	const image = (
		<img src={node.attrs?.src} width={width} height={height} alt={node.attrs?.alt} className="inline_image" />
	);

	return (
		<div>
			{node.attrs?.href ? (
				<a href={node.attrs.href} target="_blank" rel="noopener noreferrer">
					{image}
				</a>
			) : (
				<>{image}</>
			)}
		</div>
	);
};

registerNode('inline-image', (node) => <InlineImage node={node} />);
