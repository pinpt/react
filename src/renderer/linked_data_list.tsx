import { NodeProps, recurseIntoChildren, registerNode } from './register';

const LinkedDataList = ({ node }: NodeProps) => {
	return <div className="linked-data-list">{recurseIntoChildren(node)}</div>;
};

registerNode('linked_data_list', (node) => <LinkedDataList node={node} />);
