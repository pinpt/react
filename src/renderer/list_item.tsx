import { registerNode, recurseIntoChildren, NodeProps } from './register';

const ListItem = ({ node }: NodeProps) => {
	return <li>{recurseIntoChildren(node)}</li>;
};

registerNode('list_item', (node) => <ListItem node={node} />);
