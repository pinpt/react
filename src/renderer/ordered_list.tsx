import { registerNode, recurseIntoChildren, NodeProps } from './register';

const OrderedList = ({ node }: NodeProps) => {
	return <ol>{recurseIntoChildren(node)}</ol>;
};

registerNode('ordered_list', (node) => <OrderedList node={node} />);
