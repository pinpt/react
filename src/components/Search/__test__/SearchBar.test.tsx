import renderer from 'react-test-renderer';
import SearchBar from '../Bar';

test('Test default', () => {
	const component = renderer.create(<SearchBar />);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Test change handler', () => {
	const component = renderer.create(<SearchBar onChange={(val) => console.log(`value is now ${val}`)} />);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Test submit handler', () => {
	const component = renderer.create(<SearchBar onSubmit={(val) => console.log(`submitted with ${val}`)} />);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Test both handlers', () => {
	const component = renderer.create(
		<SearchBar
			onSubmit={(val) => console.log(`submitted with ${val}`)}
			onChange={(val) => console.log(`value is now ${val}`)}
		/>
	);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Test custom className', () => {
	const component = renderer.create(<SearchBar className="test-custom" />);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
