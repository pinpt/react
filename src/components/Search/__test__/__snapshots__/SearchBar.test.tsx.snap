// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test both handlers 1`] = `
<div
  className="Pinpoint Search Bar "
>
  <svg
    aria-hidden="true"
    className="icon"
    data-icon="search"
    data-prefix="fas"
    focusable="false"
    height={16}
    role="img"
    viewBox="0 0 512 512"
    width={16}
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
      fill="currentColor"
    />
  </svg>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <input
      className="input"
      onChange={[Function]}
      placeholder="Search"
      value=""
    />
  </form>
</div>
`;

exports[`Test change handler 1`] = `
<div
  className="Pinpoint Search Bar "
>
  <svg
    aria-hidden="true"
    className="icon"
    data-icon="search"
    data-prefix="fas"
    focusable="false"
    height={16}
    role="img"
    viewBox="0 0 512 512"
    width={16}
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
      fill="currentColor"
    />
  </svg>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <input
      className="input"
      onChange={[Function]}
      placeholder="Search"
      value=""
    />
  </form>
</div>
`;

exports[`Test custom className 1`] = `
<div
  className="Pinpoint Search Bar test-custom"
>
  <svg
    aria-hidden="true"
    className="icon"
    data-icon="search"
    data-prefix="fas"
    focusable="false"
    height={16}
    role="img"
    viewBox="0 0 512 512"
    width={16}
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
      fill="currentColor"
    />
  </svg>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <input
      className="input"
      onChange={[Function]}
      placeholder="Search"
      value=""
    />
  </form>
</div>
`;

exports[`Test default 1`] = `
<div
  className="Pinpoint Search Bar "
>
  <svg
    aria-hidden="true"
    className="icon"
    data-icon="search"
    data-prefix="fas"
    focusable="false"
    height={16}
    role="img"
    viewBox="0 0 512 512"
    width={16}
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
      fill="currentColor"
    />
  </svg>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <input
      className="input"
      onChange={[Function]}
      placeholder="Search"
      value=""
    />
  </form>
</div>
`;

exports[`Test submit handler 1`] = `
<div
  className="Pinpoint Search Bar "
>
  <svg
    aria-hidden="true"
    className="icon"
    data-icon="search"
    data-prefix="fas"
    focusable="false"
    height={16}
    role="img"
    viewBox="0 0 512 512"
    width={16}
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
      fill="currentColor"
    />
  </svg>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <input
      className="input"
      onChange={[Function]}
      placeholder="Search"
      value=""
    />
  </form>
</div>
`;
